src/middlewared/middlewared/plugins/smb.py

```diff
@@ -428,6 +429,15 @@ async def import_conf_to_registry(self):
             self.logger.warning('failed to load share config: %s',
                                 load.stderr.decode())
 
+    @private
```

Comment:
There is an edge-case dead lock here. So I feel like we should put a timeout on this `while True:` loop.

For example, if ctdb is started and has the `50.samba` event script enabled and, for example, the smbd service will not start, then we will deadlock here if the node that is causing the cluster to be unhealthy is the node that is running this.

A typical catch-22. So I would maybe put a timeout for 30 seconds (or whatever you think is best so that the `smb.configure` job will at least error out and give some hint on why it failed instead of deadlocking here.